onnxruntime/core/framework/debug_node_inputs_outputs_utils.cc

```diff
@@ -231,19 +240,30 @@ const NodeDumpOptions& NodeDumpOptionsFromEnvironmentVariables() {
   return node_dump_options;
 }
 
+static bool AnyOptionIsSet(const NodeDumpOptions& dump_options) {
+  return ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::ShapeOnly) != 0) ||
+         ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::InputData) != 0) ||
+         ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::OutputData) != 0);
+}
+
+#define COUT_IF(boolean_expression) \
+  if ((boolean_expression) == true) std::cout
+
 void DumpNodeInputs(
     const NodeDumpOptions& dump_options,
     const OpKernelContext& context, const Node& node, const SessionState& session_state) {
   if (!FilterNode(dump_options, node)) return;
 
-  std::cout << "-----------\n";
-  std::cout << node.OpType() << " node: " << node.Name() << "\n";
+  bool is_any_option_set = AnyOptionIsSet(dump_options);
```

Comment:
if no output is desired (`!is_any_option_set`), can we just return early? similar comment for DumpNodeOutputs()

Reply:
thanks for updating. since is_any_option_set will be true after this point, can we just unconditionally print messages other than the shape?

Reply:
Still need the PrintIf for shape for better readability.